### PR TITLE
tubep : Added text and sprite drop shadow

### PR DESCRIPTION
**Pull request by sasuke-arcade**
Text and sprite layers draw a drop shadow with 1 dot width to the left.
See the gameplay video on the PCB. https://www.youtube.com/watch?v=xxONzbUOOsw

tubep

### CODE REVIEW

**Comment by happppp** [comment]
I'd say it's a monitor artifact and not actually rendered by the arcade hardware.
Clearly, you can see it at the blue 'window' on the spaceship, there is no artifact on the left edge.

**Comment by sasuke-arcade** [comment]
If you would like to watch the video in this PR comments, please.
Clearly a shadow is being drawn.
Also, this shadow is also being drawn on the Switch and PS4 versions.

**Comment by sasuke-arcade** [comment]
Shadow will also be drawn on all enemies and "TUBE PANIC" on the title screen.
However, when text or sprites are overlapped, a shadow is drawn on the overlapped state.
So, for example, when the my ship is on the mother ship at round cleared, there is no shadow of the ship.

**Comment by happppp** [comment]
Ok I see it when player lands on mothership, indeed sprite-sprite has no artifacts.

**Comment by angelosa** [comment]
I'm not convinced this is the right behaviour: blitter class Nichibutsu devices are known to use opacity flags from the [color table lookups](https://github.com/mamedev/mame/blob/master/src/mame/video/pastelg.cpp#L221) instead. 
But then I admit the driver is hard to follow and has zero debugging aid i.e. no `m_textram` read-back in debug, no gfx decoding, not even convinced about `m_textram` not being a tilemap in the first place and so on ...

**Comment by happppp** [comment]
Since there is no artifacting(or call it shadowing) between sprite-sprite or text-sprite, I don't think it is an opaque pixel defined in the tile graphic itself. More likely, when the video hardware switches from background to sprite/text, it will have 1 NOP pixel clock, resulting in 1 black pixel.

**Comment by sasuke-arcade** [comment]
Yes, even when the sprite and the text overlap, have a constant shadowing even when sprites rotated or scaled.
Also, the PCB does not draw sprites in the left few pixels of the screen. This way the clipped sprites will also have shadowing.

clipped_sprite

**Comment by sasuke-arcade** [comment]
Just like the the tube's 3D effect or the sprite's rotation feature, the Tube Panic PCB has an innovative feature that has been added to support shorter development times.

According to an interview with Shigeki Fujiwara, the graphics designer, these were made possible by hardware engineers who "never give up on the impossible".

This shadowing is something I've never seen on other hardware, and I was skeptical at first, but given the development history, I think it doesn't seem surprising that it would have such a special feature.

**Comment by happppp** [comment]
Who knows, it could be a genius feature they came up with during rapid development. Or is it just a happy side effect with the limited hardware.

Anyway, I'm ok with the PR.

**Comment by DavidHaywood** [comment]
It reminds me a bit of the 'half pixel offset' that shadows on OutRun have, although that one looks kinda awful on real hardware once you notice it.  I'd say here it was a hardware side effect that was turned into a feature ;)

**Comment by sasuke-arcade** [comment]
Thanks for review and confirm!
